lib/bootloader_setup.pm

```diff
@@ -1003,14 +1003,15 @@ sub grep_grub_settings {
 
 =head2 grep_grub_cmdline_settings
 
-    grep_grub_cmdline_settings($pattern)
+    grep_grub_cmdline_settings($pattern [, $search])
 
 Search for C<$pattern> in grub cmdline variable (usually
 GRUB_CMDLINE_LINUX_DEFAULT) in /etc/default/grub, return 1 if found.
 =cut
 sub grep_grub_cmdline_settings {
     my $pattern = shift;
-    return grep_grub_settings(get_cmdline_var() . ".*${pattern}");
+    my $search //= get_cmdline_var();
```

Comment:
James, you miss to shift for `$search` (so `get_cmdline_var()` will be used all the time). That's the only concern.
I guess there will be always just 2 params, so we `%args` wouldn't make much sense here.

Reply:
Yes, 3+ parameters use %args make sense only. 

let me try again.